Fix compiling SDL without DBus under Linux

Kaleb-Reid opened the pull request:
Some files being compiled in the sdl driver which require DBus support do not seem to actually be used.

Fixes https://github.com/godotengine/godot/issues/111049

Kaleb-Reid committed:
Fix compiling SDL without DBus under Linux

bruvzg reviewed (commented):
Is it building and joystick input working without these files? If it is, we should remove them completely. System theme code should not be used at all, and FCITX (same for ibus) is for IME, and we only use SDL for joystick, not text input.

Kaleb-Reid commented:
@bruvzg I can build and joystick works without either of these files, so I've just removed them. I also removed two other files related to IME which don't appear to do anything.

I do get a bug when compiling with ```threads=no``` which causes input to not work (can't click anything on the project selection screen) and
```
WARNING: Keeping screen on not supported by this display server.
     at: screen_set_keep_on (servers/display_server.cpp:579)
```
to be printed to the console, but my assumption is that this doesn't have anything to do with this pr.

akien-mga reviewed (approved):
Looks good to me, always nice when a diff is:
<img width="127" height="38" alt="image" src="https://github.com/user-attachments/assets/3f54f887-c95b-4c25-a495-200b59145bc3" />
